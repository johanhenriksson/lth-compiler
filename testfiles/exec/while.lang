int main() {
    int i = 0;

    while(i < 3) {
        i = i + 1;
    }
    assert(i == 3);

    while (i < 3) {
        i = i + 99;
    }
    assert(i == 3);

    return 0;
}
