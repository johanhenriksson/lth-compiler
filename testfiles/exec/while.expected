; Library Functions
declare i8* @__malloc(i32)
declare void @__delete(i8*)
declare void @print(i64)
declare i64 @read()
declare i64 @truncate(double)
declare double @promote(i64)
declare void @assert(i1)
declare void @except(i1)

define i64 @main() {
%i = alloca i64
store i64 0, i64* %i
br label %while1
while1:
; IdUse i
%_2 = load i64, i64* %i
%_1 = icmp slt i64 %_2, 3
br i1 %_1, label %while1_loop, label %while1_end
while1_loop:
; IdUse i
%_4 = load i64, i64* %i
%_3 = add i64 %_4, 1
store i64 %_3, i64* %i
br label %while1
while1_end:
; IdUse i
%_6 = load i64, i64* %i
%_5 = icmp eq i64 %_6, 3
call void @assert(i1 %_5)
br label %while2
while2:
; IdUse i
%_8 = load i64, i64* %i
%_7 = icmp slt i64 %_8, 3
br i1 %_7, label %while2_loop, label %while2_end
while2_loop:
; IdUse i
%_10 = load i64, i64* %i
%_9 = add i64 %_10, 99
store i64 %_9, i64* %i
br label %while2
while2_end:
; IdUse i
%_12 = load i64, i64* %i
%_11 = icmp eq i64 %_12, 3
call void @assert(i1 %_11)
ret i64 0
}

